content/docs/release-notes/release-notes-1.12.md

```diff
@@ -3,8 +3,102 @@ title: Release 1.12
+automatically adds this label to all `Certificate` `Secret`s.
+
+This change has been placed behind alpha feature gate as it could potentially
+slow down large scale issuance because issuer credentials `Secret`s will now be
+retrieved from kube apiserver instead of local cache. To prevent the slow down,
+users can manually label issuer `Secret`s with a
+`controller.cert-manager.io/fao` label.
+See the
+[design](https://github.com/cert-manager/cert-manager/blob/master/design/20221205-memory-management.md)
+and [implementation](https://github.com/cert-manager/cert-manager/pull/5824) for
+additional details.
+We would like to gather some feedback on this change before
+it can graduate- please leave your comments on
+(`cert-manager#6074`)[https://github.com/cert-manager/cert-manager/issues/6074].
+
+Additionally, controller no longer watches and caches all `Pod` and `Service`
+resources.
+See [`cert-manager#5976`](https://github.com/cert-manager/cert-manager/pull/5976) for implementation.
+
+#### Cainjector
+
+[Cainjector's](../concepts/ca-injector.md) control loops have been refactored, so by default it should
+consume up to twice less memory, see
+[`cert-manager#5746`](https://github.com/cert-manager/cert-manager/pull/5746).
+
+Additionally, a number of flags have been added to cainjector that can be used
+to scope down what resources it watches and caches.
+
+If cainjector is only used as part of cert-manager installation, it only needs
+to inject CA certs to cert-manager's `MutatingWebhookConfiguration` and
+`ValidatingWebhookConfiguration` from a `Secret` in cert-manager's installation
+namespace so all the other injectable/source types can be turned off and
+cainjector can be scoped to a single namespace, see the relevant flags below:
+
+```go
+// cainjector flags
+--namespace=<cert-manager-installation-namespace> \
+--enable-customresourcedefinitions-injectable=false \
+--enable-certificates-data-source=false \
+--enable-apiservices-injectable=false
+```
+
+See [`cert-manager#5766`](https://github.com/cert-manager/cert-manager/pull/5766) for more detail.
+
+A big thanks to everyone who put in time reporting and writing up issues
+describing performance problems in large scale installations.
+
+### Improved Security with Independent Go Modules for Each Binary
+
+With cert-manager 1.12, we have made significant changes aimed at improving our
+reaction time to vulnerability reports. Each binary now has its own `go.mod`
+file, providing us with more flexibility to react to CVEs.
+
+In the past, we have been unable to offer security patches due to unsupported
+dependencies. An example of this was seen with Helm: in cert-manager 1.10, we
+were unable to fix a CVE reported in Helm because Helm only offers security
+patches for its latest minor version.
+
+While this doesn't decrease the dependency attack surface for any of the
+binaries, it does allow us to react more quickly and effectively when a
+vulnerability is reported.
```

Comment:
suggestion: Would it be worth coming at this from a slightly different angle?

```markdown
### Reduced Transitive Dependencies in cert-manager

cert-manager 1.12 separates some of its functionality into separate Go modules,
with the aim of decreasing the amount of time and toil for maintainers to patch
vulnerable dependencies.

A side effect of this is that it's now easier to import cert-manager in Go,
in terms of transitive dependencies that might show up in your `go.mod` files or
potential version conflicts between cert-manager and your other dependencies.

The caveat here is that we still only recommend importing cert-manager in
[very specific circumstances](../contributing/importing.md), and the module changes
mean that if you imported some paths (specifically under `cmd` or some paths under `test`)
you might see broken imports when you try to upgrade.

If you experience a break as part of this, we're sorry and we'd be interested to chat about it.
The vast majority of projects using cert-manager should notice no impact, and there should
be no runtime impact either.
```

Reply:
> ```md
> ##### Reduced Transitive Dependencies in cert-manager
> ```

The title "Reduced Transitive Dependencies in cert-manager" doesnt state the benefit that end-users can expect from that change.

> cert-manager 1.12 separates some of its functionality into separate Go modules,
> with the aim of decreasing the amount of time and toil for maintainers to patch
> vulnerable dependencies.

I think first paragraph should focus on the problem that we have been solving: reducing the impacts of breaking the rule "no minor bumps in patches" when a CVE is declared in an unsupported minor version of a dependency. Hence, we are increasing (or sustaining) the security of the end-user installations.

> ```md
> A side effect of this is that it's now easier to import cert-manager in Go,
> in terms of transitive dependencies that might show up in your `go.mod` files or
> potential version conflicts between cert-manager and your other dependencies.
> 
> The caveat here is that we still only recommend importing cert-manager in
> [very specific circumstances](../contributing/importing.md), and the module changes
> mean that if you imported some paths (specifically under `cmd` or some paths under `test`)
> you might see broken imports when you try to upgrade.
> 
> If you experience a break as part of this, we're sorry and we'd be interested to chat about it.
> The vast majority of projects using cert-manager should notice no impact, and there should
> be no runtime impact either.
> ```

These three paragraph look great, I hadn't thought about these.

Reply:
I added the three paragraphs and re-wrote my paragraphs. Please take another look

Reply:
> These three paragraph look great, I hadn't thought about these.

I did a _really_ awful job of communicating this change, wow

Reply:
New text looks fine!